[23075][23087] Force selection of values when select is required

oliverguenther opened the pull request:
This changes the disabled default option for required select fields to select that by default and disable it.
This causes a required field to have a default selection of `Please select` similar to the Rails selects.

![bildschirmfoto 2016-05-04 um 08 31 16](https://cloud.githubusercontent.com/assets/459462/15006617/ac029fcc-11d2-11e6-89f2-7b12adbe4e90.png)

![bildschirmfoto 2016-05-04 um 08 31 21](https://cloud.githubusercontent.com/assets/459462/15006616/aa95f12a-11d2-11e6-8888-ad8f9a6c3f16.png)

https://community.openproject.com/work_packages/23075/activity

---

Angular doesn't seem to respect the string 'null' option correctly when
removing an assigned field.

Instead, the model is set to undefined and we need to check for that.

https://community.openproject.com/work_packages/23087/activity

oliverguenther committed:
Force selection of values when select is required

TeatroIO commented:
I've prepared a stage to preview changes. <a href='http://oliverguenther-fix-23075-first-select-value.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/' target='_blank'>Open stage</a> or <a href='http://oliverguenther-fix-23075-first-select-value.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com//teatro.log' target='_blank'>view logs</a>.

oliverguenther committed:
Fix default value of null

Angular doesn't seem to respect the string 'null' option correctly when
removing an assigned field.

Instead, the model is set to undefined and we need to check for that.

https://community.openproject.com/work_packages/23087/activity

oliverguenther committed:
Create spec for required custom field list and text

oliverguenther commented:
Rebased to run on https://github.com/opf/openproject/pull/4338

oliverguenther committed:
Do not store allowedValue when it is empty

If the user may edit a work package but is not a member of that project
or has no workflow (e.g., Admin without role in that project), it may
start editing a workpackage, but fail to save it due to the status being
empty, but required.

We should not copy the allowed value of a property when its embedded schema doesn't allow changing it.

oliverguenther committed:
Show empty -- Please Select -- only when no value present
